h2o-core/src/main/java/water/rapids/ast/prims/advmath/SpearmanCorrelation.java

```diff
@@ -339,8 +340,22 @@ public double getSpearmanCorrelationCoefficient() {
       }
     }
 
-    return new MeanTask()
-            .doAll(vecs)._means;
+    if (VecUtils.areVectorsMutuallyCompatible(vecs)) {
+      return new MeanTask()
+              .doAll(vecs)._means;
+    } else {
```

Comment:
this is not necessary

new Frame().add(null, vecs) will make the vecs compatible

Reply:
1. It would be best for me to return the vecs already compatible from `rankedVectors` method.
2. (note to self) Need to return the vecs from inside the Frame, as it might end up duplicating the vector given internally, creating a new one.